### PR TITLE
feat: add certificates for ssl test and an ssl test image

### LINKED ISSUE #159: [Question] Testing of custom certificates
We're supporting custom certificates for connection, which is a great feature. But testing whether it works is not as trivial.

For example, I am trying to implement custom certs in Python. I want to test this and ensure this feature works as expected with a custom cert. However, I can't do so, as we don't have the appropriate test setup.

I opened this issue to discuss approaches and what we can think of doing to ensure this feature works as intended.
My proposal (not sure if it is feasible) would be:
- creating an own root CA
- setting up an additional flagd-testbed image with just this one root.ca
- Provide a certificate within the test-harness and use this in e2e tests for proper testing of the custom SSL certificate behavior.
- We do not not even need to provide another gherkin file for this feature, we could simply create it with the existing ones

Question:
- do you think my approach is feasible?
- do you think we should provide such a docker image?
- is it worth to put the effort into such a testing setup?

### PR DESCRIPTION
**Pull request by aepfli**
This adds another image protected via an SSL certificate with a custom CA.

If a key is used, the password is `flagd` in the creation process.

### dockerfile improvements

Instead of having now two different dockerfiles, copying the files, and executing the script, i reduced one layer, by depending on the _testbed_ image and only changing the entry point

### testing

first run the docker container `docker run --rm -p 8013:8013 flagd-testbed-ssl`

```curl -v --cacert ./custom-root-cert.crt https://localhost:8013/health``` should run in to 404

```curl -v https://localhost:8013/health``` should run in to a cert error

closes #159

### CODE REVIEW

**Comment by aepfli** [comment]
links to https://github.com/open-feature/flagd-testbed/issues/159

**Comment by aepfli** [comment]
tested with java, worked

**Comment by aepfli** [comment]
the only thing which we might want to add, is also a gherkin file for testing, but on the other hand, we could reuse the evaluation.feature with the basic testing flags

File: ssl/custom-ca.key [hunk 1]
```diff
@@ -0,0 +1,52 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (beeme1mr) [inline ssl/custom-ca.key:1]: Should we make this a GitHub secret? I know it's just for testing but I'm afraid scanners might pick this up and open issues in the repo.

**Comment by toddbaert** [comment]
:clap: :clap: :clap: :clap: 

Can't wait to have this implemented in various places. I'd be happy to help.

**Review by toddbaert — APPROVED** [review]

**Pull request merged**